strings/base_collections_base.h

```diff
@@ -370,24 +488,33 @@ WINRT_EXPORT namespace winrt
     {
         V Lookup(K const& key) const
         {
-            auto pair = static_cast<D const&>(*this).get_container().find(static_cast<D const&>(*this).wrap_value(key));
-
-            if (pair == static_cast<D const&>(*this).get_container().end())
+            return static_cast<D const&>(*this).perform_shared([&]
             {
-                throw hresult_out_of_bounds();
-            }
+                auto pair = static_cast<D const&>(*this).get_container().find(static_cast<D const&>(*this).wrap_value(key));
 
-            return static_cast<D const&>(*this).unwrap_value(pair->second);
+                if (pair == static_cast<D const&>(*this).get_container().end())
+                {
+                    throw hresult_out_of_bounds();
+                }
+
+                return static_cast<D const&>(*this).unwrap_value(pair->second);
+            });
         }
 
         uint32_t Size() const noexcept
         {
-            return static_cast<uint32_t>(static_cast<D const&>(*this).get_container().size());
+            return static_cast<D const&>(*this).perform_shared([&]
+            {
+                return static_cast<uint32_t>(static_cast<D const&>(*this).get_container().size());
+            });
         }
 
         bool HasKey(K const& key) const noexcept
         {
-            return static_cast<D const&>(*this).get_container().find(static_cast<D const&>(*this).wrap_value(key)) != static_cast<D const&>(*this).get_container().end();
+            return static_cast<D const&>(*this).perform_shared([&]
+            {
+                return static_cast<D const&>(*this).get_container().find(static_cast<D const&>(*this).wrap_value(key)) != static_cast<D const&>(*this).get_container().end();
+            });
         }
 
         void Split(Windows::Foundation::Collections::IMapView<K, V>& first, Windows::Foundation::Collections::IMapView<K, V>& second) const noexcept
```

Comment:
I never did figure out what Split does.

Reply:
It does nothing. In theory it was meant to split the container for parallel search algorithms. That wasn't a well thought out plan. Fortunately a "valid" implementation can do nothing.